### PR TITLE
HDDS-2166. Some RPC metrics are missing from SCM prometheus endpoint

### PR DESCRIPTION
**Pull request by elek**
In Hadoop metrics it's possible to register multiple metrics with the same name but with different tags. For example each RpcServere has an own metrics instance in SCM.

{code}
    "name" : "Hadoop:service=StorageContainerManager,name=RpcActivityForPort9860",
    "name" : "Hadoop:service=StorageContainerManager,name=RpcActivityForPort9863",
{code}

They are converted by PrometheusSink to a prometheus metric line with proper name and tags. For example:

{code}
rpc_rpc_queue_time60s_num_ops{port="9860",servername="StorageContainerLocationProtocolService",context="rpc",hostname="72736061cbc5"} 0
{code}

The PrometheusSink uses a Map to cache all the recent values but unfortunately the key contains only the name (rpc_rpc_queue_time60s_num_ops in our example) but not the tags (port=...)

For this reason if there are multiple metrics with the same name, only the first one will be displayed.

As a result in SCM only the metrics of the first RPC server can be exported to the prometheus endpoint. 

See: https://issues.apache.org/jira/browse/HDDS-2166

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 40 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 50 | hadoop-hdds in trunk failed. |
| -1 | mvninstall | 32 | hadoop-ozone in trunk failed. |
| -1 | compile | 21 | hadoop-hdds in trunk failed. |
| -1 | compile | 16 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 67 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 869 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 22 | hadoop-hdds in trunk failed. |
| -1 | javadoc | 20 | hadoop-ozone in trunk failed. |
| 0 | spotbugs | 964 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 32 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 17 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 33 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 28 | hadoop-ozone in the patch failed. |
| -1 | compile | 24 | hadoop-hdds in the patch failed. |
| -1 | compile | 19 | hadoop-ozone in the patch failed. |
| -1 | javac | 24 | hadoop-hdds in the patch failed. |
| -1 | javac | 19 | hadoop-ozone in the patch failed. |
| +1 | checkstyle | 57 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 709 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 22 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 19 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 31 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 20 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 28 | hadoop-hdds in the patch failed. |
| -1 | unit | 23 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 34 | The patch does not generate ASF License warnings. |
| | | 2352 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1505 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 8f414d43ab07 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 2b5fc95 |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/branch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/patch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/testReport/ |
| Max. process+thread count | 447 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/framework U: hadoop-hdds/framework |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 38 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 29 | hadoop-hdds in trunk failed. |
| -1 | mvninstall | 30 | hadoop-ozone in trunk failed. |
| -1 | compile | 21 | hadoop-hdds in trunk failed. |
| -1 | compile | 14 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 60 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 859 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 23 | hadoop-hdds in trunk failed. |
| -1 | javadoc | 20 | hadoop-ozone in trunk failed. |
| 0 | spotbugs | 964 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 34 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 21 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 35 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 32 | hadoop-ozone in the patch failed. |
| -1 | compile | 25 | hadoop-hdds in the patch failed. |
| -1 | compile | 19 | hadoop-ozone in the patch failed. |
| -1 | javac | 25 | hadoop-hdds in the patch failed. |
| -1 | javac | 19 | hadoop-ozone in the patch failed. |
| +1 | checkstyle | 57 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 697 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 23 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 18 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 31 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 16 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 27 | hadoop-hdds in the patch failed. |
| -1 | unit | 20 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 34 | The patch does not generate ASF License warnings. |
| | | 2301 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1505 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux e88a1b57f239 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 51c64b3 |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/branch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/patch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/testReport/ |
| Max. process+thread count | 462 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/framework U: hadoop-hdds/framework |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/2/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 39 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 38 | hadoop-hdds in trunk failed. |
| -1 | mvninstall | 27 | hadoop-ozone in trunk failed. |
| -1 | compile | 21 | hadoop-hdds in trunk failed. |
| -1 | compile | 12 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 62 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 916 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 17 | hadoop-hdds in trunk failed. |
| -1 | javadoc | 15 | hadoop-ozone in trunk failed. |
| 0 | spotbugs | 997 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 28 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 16 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 32 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 27 | hadoop-ozone in the patch failed. |
| -1 | compile | 22 | hadoop-hdds in the patch failed. |
| -1 | compile | 15 | hadoop-ozone in the patch failed. |
| -1 | javac | 22 | hadoop-hdds in the patch failed. |
| -1 | javac | 15 | hadoop-ozone in the patch failed. |
| +1 | checkstyle | 54 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 772 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 17 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 16 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 28 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 15 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 26 | hadoop-hdds in the patch failed. |
| -1 | unit | 20 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 30 | The patch does not generate ASF License warnings. |
| | | 2376 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1505 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 71aff0ab79f9 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c724577 |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/branch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/patch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/testReport/ |
| Max. process+thread count | 306 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/framework U: hadoop-hdds/framework |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/3/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/server/TestPrometheusMetricsSink.java [hunk 1]
```diff
@@ -71,6 +73,50 @@ public void testPublish() throws IOException {
...
+    ByteArrayOutputStream stream = new ByteArrayOutputStream();
+    OutputStreamWriter writer = new OutputStreamWriter(stream, UTF_8);
+
+    //WHEN
+    sink.writeMetrics(writer);
+    writer.flush();
+
+    //THEN
+    String writtenMetrics = stream.toString(UTF_8.name());
+    System.out.println(writtenMetrics);
```
Comment (xiaoyuyao) [inline hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/server/TestPrometheusMetricsSink.java]: NIT: debug message can be removed.

**Comment by elek** [comment]
Thanks @xiaoyuyao the review. I will commit it after a new jenkins build.

Debug System.out.println is removed (from the patch and also from the previous method).

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 35 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 33 | hadoop-hdds in trunk failed. |
| -1 | mvninstall | 26 | hadoop-ozone in trunk failed. |
| -1 | compile | 21 | hadoop-hdds in trunk failed. |
| -1 | compile | 14 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 53 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 957 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 18 | hadoop-hdds in trunk failed. |
| -1 | javadoc | 18 | hadoop-ozone in trunk failed. |
| 0 | spotbugs | 1045 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 30 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 18 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 34 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 27 | hadoop-ozone in the patch failed. |
| -1 | compile | 23 | hadoop-hdds in the patch failed. |
| -1 | compile | 15 | hadoop-ozone in the patch failed. |
| -1 | javac | 23 | hadoop-hdds in the patch failed. |
| -1 | javac | 15 | hadoop-ozone in the patch failed. |
| +1 | checkstyle | 60 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 800 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 18 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 16 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 28 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 17 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 24 | hadoop-hdds in the patch failed. |
| -1 | unit | 19 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 28 | The patch does not generate ASF License warnings. |
| | | 2447 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1505 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 898aada9c0c8 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 8a9ede5 |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/branch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/patch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/testReport/ |
| Max. process+thread count | 341 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/framework U: hadoop-hdds/framework |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1505/4/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**